djangosaml2/views.py

```diff
@@ -145,7 +145,10 @@ def login(request,
     #
     # Read more in the official SAML2 specs (3.4.4.1):
     # http://docs.oasis-open.org/security/saml/v2.0/saml-bindings-2.0-os.pdf
-    binding = BINDING_HTTP_POST if conf._sp_authn_requests_signed else BINDING_HTTP_REDIRECT
+    binding = BINDING_HTTP_REDIRECT
+    if hasattr(conf, '_sp_authn_requests_signed') and \
+            conf._sp_authn_requests_signed:
+        binding = BINDING_HTTP_POST
```

Comment:
I agree with getattr over hasattr, thanks!

I'd prefer a nested if over a one-liner in this case, as I try to keep my lines under 80 to fit PEP8. Anyway since this file has other lines with this style I have no problem of re-writing it as you said.

I'll commit right now.